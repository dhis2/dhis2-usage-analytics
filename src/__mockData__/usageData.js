export const dataStatistics = [
    {
        year: 2016,
        week: 22,
        mapViews: 4457,
        chartViews: 5047,
        pivotTableViews: 4281,
        eventReportViews: 4649,
        eventChartViews: 4716,
        dashboardViews: 6460,
        dataSetReportViews: 4414,
        totalViews: 23669,
        averageViews: 104,
        averageMapViews: 19,
        averageChartViews: 22,
        averagePivotTableViews: 18,
        averageEventReportViews: 20,
        averageEventChartViews: 20,
        averageDashboardViews: 28,
        savedMaps: 1658,
        savedCharts: 1723,
        savedPivotTables: 1642,
        savedEventReports: 1748,
        savedEventCharts: 1613,
        savedDashboards: 1873,
        savedIndicators: 1826,
        savedDataValues: 11751,
        activeUsers: 227,
        users: 262,
    },
    {
        year: 2016,
        week: 23,
        mapViews: 4093,
        chartViews: 5179,
        pivotTableViews: 4275,
        eventReportViews: 4856,
        eventChartViews: 4782,
        dashboardViews: 6193,
        dataSetReportViews: 4242,
        totalViews: 23840,
        averageViews: 101,
        averageMapViews: 17,
        averageChartViews: 22,
        averagePivotTableViews: 18,
        averageEventReportViews: 20,
        averageEventChartViews: 20,
        averageDashboardViews: 26,
        savedMaps: 1593,
        savedCharts: 1668,
        savedPivotTables: 1643,
        savedEventReports: 1745,
        savedEventCharts: 1672,
        savedDashboards: 1853,
        savedIndicators: 1905,
        savedDataValues: 11520,
        activeUsers: 234,
        users: 262,
    },
]

export const topFavorites = [
    {
        position: 1,
        name: 'ANC: ANC 3 coverage this year (gauge)',
        views: 1,
        id: 'dw62KezChat',
        created: '2014-11-10T00:00:00.000',
    },
    {
        position: 2,
        name: 'Immunization: IPT 1 and 2 Coverage',
        views: 1,
        id: 'nwqNUFG9Vky',
        created: '2012-07-31T00:00:00.000',
    },
]
