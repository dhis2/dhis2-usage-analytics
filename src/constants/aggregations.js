import i18n from '@dhis2/d2-i18n'

export const SUM = 'SUM'
export const AVERAGE = 'AVERAGE'

const AGGREGATIONS = {
    [SUM]: { value: SUM, label: i18n.t('Sum') },
    [AVERAGE]: { value: AVERAGE, label: i18n.t('Average') },
}
export default AGGREGATIONS
