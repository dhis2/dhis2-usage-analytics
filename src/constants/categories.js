import i18n from '@dhis2/d2-i18n'

export const FAVORITE_VIEWS = 'FAVORITE_VIEWS'
export const FAVORITES_SAVED = 'FAVORITES_SAVED'
export const USERS = 'USERS'
export const TOP_FAVORITES = 'TOP_FAVORITES'
export const DATA_VALUES = 'DATA_VALUES'

const CATEGORIES = [
    { value: FAVORITE_VIEWS, label: i18n.t('Favorite views') },
    { value: FAVORITES_SAVED, label: i18n.t('Favorites saved') },
    { value: USERS, label: i18n.t('Users') },
    { value: TOP_FAVORITES, label: i18n.t('Top favorites') },
    { value: DATA_VALUES, label: i18n.t('Data values') },
]
export default CATEGORIES
