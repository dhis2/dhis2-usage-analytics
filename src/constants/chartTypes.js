import i18n from '@dhis2/d2-i18n'

export const ALL = 'ALL'
export const TOTAL = 'TOTAL'

const chartTypes = [
    { value: ALL, label: i18n.t('All') },
    { value: TOTAL, label: i18n.t('Total') },
]

export default chartTypes
