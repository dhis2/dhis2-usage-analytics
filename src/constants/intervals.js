import i18n from '@dhis2/d2-i18n'

export const DAY = 'DAY'
export const WEEK = 'WEEK'
export const MONTH = 'MONTH'
export const YEAR = 'YEAR'

const intervals = [
    { value: DAY, label: i18n.t('Day') },
    { value: WEEK, label: i18n.t('Week') },
    { value: MONTH, label: i18n.t('Month') },
    { value: YEAR, label: i18n.t('Year') },
]

export default intervals
