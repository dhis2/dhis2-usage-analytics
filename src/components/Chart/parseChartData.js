import { CHART_FIELDS as FIELDS } from '../../constants/fields'
import CATEGORIES, { FAVORITE_VIEWS } from '../../constants/categories'
import { getDisplayDateForInterval } from '../../utils/date'
import * as config from './config'

// This will ensure that the chart at has an yAxis from 0 - 1
const MIN_SUGGESTED_MAX_VALUE = 1

export default function parseChartData(
    { aggregationLevel, category, chartType, interval },
    dataPoints
) {
    const labels = []
    const datasets = []
    const title = CATEGORIES[category].label
    const subtitle = CATEGORIES[category].subtitle
    let max = MIN_SUGGESTED_MAX_VALUE
    let min = null
    const fields =
        category === FAVORITE_VIEWS
            ? FIELDS[category][aggregationLevel][chartType]
            : FIELDS[category]
    const dataPointsLen = dataPoints.length
    const fieldsLen = fields.length

    for (
        let dataPointIndex = 0;
        dataPointIndex < dataPointsLen;
        dataPointIndex++
    ) {
        const dataPoint = dataPoints[dataPointIndex]
        labels.push(getDisplayDateForInterval(dataPoint, interval))

        for (let fieldIndex = 0; fieldIndex < fieldsLen; fieldIndex++) {
            const field = fields[fieldIndex]

            if (dataPointIndex === 0) {
                datasets.push({
                    ...config.dataSet,
                    label: field.label,
                    data: [],
                    borderColor: config.colors[fieldIndex],
                })
            }

            const pointValue = dataPoint[field.key]

            if (pointValue > max) {
                max = pointValue
            }

            if (min === null || pointValue < min) {
                min = pointValue
            }

            datasets[fieldIndex].data.push(pointValue)
        }
    }

    config.options.scales.yAxes[0].ticks.suggestedMax = max
    config.options.scales.yAxes[0].ticks.suggestedMin = min

    config.options.onResize = onChartResize

    return {
        data: {
            labels,
            datasets,
        },
        options: config.options,
        title,
        subtitle,
    }
}

export function onChartResize(chart) {
    chart.resize()
}
