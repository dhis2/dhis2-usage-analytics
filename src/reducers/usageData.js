import { LOADING, ERROR } from '../constants/statuses'
import {
    APP_LOAD_SUCCESS,
    USAGE_DATA_REQUESTED,
    USAGE_DATA_RECEIVED,
    USAGE_DATA_ERRORED,
} from '../actions/types'

export default function usageData(state = LOADING, { type, payload }) {
    switch (type) {
        case USAGE_DATA_REQUESTED:
            return LOADING
        case APP_LOAD_SUCCESS:
            return payload
        case USAGE_DATA_RECEIVED:
            return payload
        case USAGE_DATA_ERRORED:
            return ERROR
        default:
            return state
    }
}
