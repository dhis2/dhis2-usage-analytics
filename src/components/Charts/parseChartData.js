import { CHART_FIELDS as FIELDS } from './fields.js'
import { FAVORITE_VIEWS } from '../../constants/categories'
import { formatIntervalDate } from '../../utils/date.js'
import * as config from './config'

// This will ensure that the chart at has an yAxis from 0 - 1
const MIN_SUGGESTED_MAX_VALUE = 1

export default function parseChartData({
    aggregation,
    category,
    chartType,
    interval,
    data,
}) {
    const labels = []
    const datasets = []
    let max = MIN_SUGGESTED_MAX_VALUE
    let min = null
    const fields =
        category === FAVORITE_VIEWS
            ? FIELDS[category][aggregation][chartType]
            : FIELDS[category]
    const dataPointsLen = data.length
    const fieldsLen = fields.length

    for (
        let dataPointIndex = 0;
        dataPointIndex < dataPointsLen;
        dataPointIndex++
    ) {
        const dataPoint = data[dataPointIndex]
        labels.push(formatIntervalDate(dataPoint, interval))

        for (let fieldIndex = 0; fieldIndex < fieldsLen; fieldIndex++) {
            const field = fields[fieldIndex]

            if (dataPointIndex === 0) {
                datasets.push({
                    ...config.dataSet,
                    label: field.label,
                    data: [],
                    borderColor: config.colors[fieldIndex],
                })
            }

            const pointValue = dataPoint[field.key]

            if (pointValue > max) {
                max = pointValue
            }

            if (min === null || pointValue < min) {
                min = pointValue
            }

            datasets[fieldIndex].data.push(pointValue)
        }
    }

    config.options.scales.yAxes[0].ticks.suggestedMax = max
    config.options.scales.yAxes[0].ticks.suggestedMin = min

    config.options.onResize = onChartResize

    return {
        data: {
            labels,
            datasets,
        },
        options: config.options,
    }
}

export function onChartResize(chart) {
    chart.resize()
}
