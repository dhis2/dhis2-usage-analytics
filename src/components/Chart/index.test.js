import React from 'react'
import { shallow } from 'enzyme'
import { Chart, mapStateToProps } from '.'
import { onChartResize } from './parseChartData'
import { LOADING } from '../../constants/statuses'
import * as CATS from '../../constants/categories'
import * as AGGRS from '../../constants/aggregations'
import * as CHARTS from '../../constants/chartTypes'
import { WEEK } from '../../constants/intervals'
import { dataStatistics } from '../../__mockData__/usageData'

const defaultState = {
    filter: {
        aggregationLevel: AGGRS.SUM,
        category: CATS.FAVORITE_VIEWS,
        interval: WEEK,
        chartType: CHARTS.TOTAL,
    },
    usageData: dataStatistics,
}

const getWrapperForState = ({ filter, usageData }) => {
    const state = {
        filter: { ...defaultState.filter, ...filter },
        usageData: usageData || defaultState.usageData,
    }
    const props = mapStateToProps(state)
    return shallow(<Chart {...props} />)
}

describe('<Chart/>', () => {
    it('Renders a <CircularLoader/> when usageData equals LOADING', () => {
        const wrapper = getWrapperForState({ usageData: LOADING })
        expect(wrapper.find('CircularLoader').length).toEqual(1)
    })
    it('Renders nothing when category equals TOP_FAVORITES', () => {
        const wrapper = getWrapperForState({
            filter: { category: CATS.TOP_FAVORITES },
        })
        expect(wrapper).toBeEmptyRender()
    })
    it('Produces a correct chart for category FAVORITE_VIEWS, aggregationLevel SUM, and chartType TOTAL', () => {
        const wrapper = getWrapperForState({})
        expect(wrapper).toMatchSnapshot()
    })
    it('Produces a correct chart for category FAVORITE_VIEWS, aggregationLevel SUM, and chartType ALL', () => {
        const wrapper = getWrapperForState({
            filter: { chartType: CHARTS.ALL },
        })
        expect(wrapper).toMatchSnapshot()
    })
    it('Produces a correct chart for category FAVORITE_VIEWS, aggregationLevel AVERAGE, and chartType TOTAL', () => {
        const wrapper = getWrapperForState({
            filter: { aggregationLevel: AGGRS.AVERAGE },
        })
        expect(wrapper).toMatchSnapshot()
    })
    it('Produces a correct chart for category FAVORITE_VIEWS, aggregationLevel AVERAGE, and chartType ALL', () => {
        const wrapper = getWrapperForState({
            filter: { aggregationLevel: AGGRS.AVERAGE, chartType: CHARTS.ALL },
        })
        expect(wrapper).toMatchSnapshot()
    })
    it('Produces a correct chart for category FAVORITES_SAVED', () => {
        const wrapper = getWrapperForState({
            filter: { category: CATS.FAVORITES_SAVED },
        })
        expect(wrapper).toMatchSnapshot()
    })
    it('Produces a correct chart for category USERS', () => {
        const wrapper = getWrapperForState({ filter: { category: CATS.USERS } })
        expect(wrapper).toMatchSnapshot()
    })
    it('Produces a correct chart for category DATA_VALUES', () => {
        const wrapper = getWrapperForState({
            filter: { category: CATS.DATA_VALUES },
        })
        expect(wrapper).toMatchSnapshot()
    })
})

describe('onChartResize', () => {
    it('calls the resize method on the chart object', () => {
        const mockResize = jest.fn()
        const mockChart = {
            resize: mockResize,
        }
        onChartResize(mockChart)
        expect(mockResize).toHaveBeenCalledTimes(1)
    })
})
