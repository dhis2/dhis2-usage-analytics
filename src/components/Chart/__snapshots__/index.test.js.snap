// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chart/> Produces a correct chart for category DATA_VALUES 1`] = `
<div
  className="uua-data-container uua-chart-container"
>
  <h4
    className="uaa-chart-title"
  >
    Data values
  </h4>
  <h6
    className="uaa-chart-subtitle"
  >
    Number of data values saved in the system
  </h6>
  <div
    className="uaa-chart-wrap"
  >
    <Line
      data={
        Object {
          "datasets": Array [
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "borderColor": "#7cb5ec",
              "data": Array [
                11751,
                11520,
              ],
              "fill": false,
              "label": "Data Values",
              "lineTension": 0,
            },
          ],
          "labels": Array [
            "Week 22 / 2016",
            "Week 23 / 2016",
          ],
        }
      }
      options={
        Object {
          "animation": Object {
            "duration": 180,
          },
          "legend": Object {
            "position": "bottom",
          },
          "maintainAspectRatio": false,
          "onResize": [Function],
          "responsive": true,
          "scales": Object {
            "xAxes": Array [
              Object {
                "display": true,
                "scaleLabel": Object {
                  "display": true,
                },
                "ticks": Object {
                  "maxRotation": 45,
                  "minRotation": 45,
                },
              },
            ],
            "yAxes": Array [
              Object {
                "display": true,
                "scaleLabel": Object {
                  "display": true,
                  "labelString": "Values",
                },
                "ticks": Object {
                  "precision": 0,
                  "suggestedMax": 11751,
                  "suggestedMin": 11520,
                },
              },
            ],
          },
          "tooltips": Object {
            "mode": "nearest",
          },
        }
      }
    />
  </div>
</div>
`;

exports[`<Chart/> Produces a correct chart for category FAVORITE_VIEWS, aggregationLevel AVERAGE, and chartType ALL 1`] = `
<div
  className="uua-data-container uua-chart-container"
>
  <h4
    className="uaa-chart-title"
  >
    Favorite views
  </h4>
  <h6
    className="uaa-chart-subtitle"
  >
    Number of times users looked at analysis favorites
  </h6>
  <div
    className="uaa-chart-wrap"
  >
    <Line
      data={
        Object {
          "datasets": Array [
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "borderColor": "#7cb5ec",
              "data": Array [
                19,
                17,
              ],
              "fill": false,
              "label": "Average Map",
              "lineTension": 0,
            },
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "borderColor": "#CC6600",
              "data": Array [
                22,
                22,
              ],
              "fill": false,
              "label": "Average Chart",
              "lineTension": 0,
            },
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "borderColor": "#CCCC00",
              "data": Array [
                18,
                18,
              ],
              "fill": false,
              "label": "Average Pivot Table",
              "lineTension": 0,
            },
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "borderColor": "#66CC00",
              "data": Array [
                20,
                20,
              ],
              "fill": false,
              "label": "Average Event Report",
              "lineTension": 0,
            },
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "borderColor": "#ff0066",
              "data": Array [
                20,
                20,
              ],
              "fill": false,
              "label": "Average Event Chart",
              "lineTension": 0,
            },
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "borderColor": "#000000",
              "data": Array [
                28,
                26,
              ],
              "fill": false,
              "label": "Average Dashboard",
              "lineTension": 0,
            },
          ],
          "labels": Array [
            "Week 22 / 2016",
            "Week 23 / 2016",
          ],
        }
      }
      options={
        Object {
          "animation": Object {
            "duration": 180,
          },
          "legend": Object {
            "position": "bottom",
          },
          "maintainAspectRatio": false,
          "onResize": [Function],
          "responsive": true,
          "scales": Object {
            "xAxes": Array [
              Object {
                "display": true,
                "scaleLabel": Object {
                  "display": true,
                },
                "ticks": Object {
                  "maxRotation": 45,
                  "minRotation": 45,
                },
              },
            ],
            "yAxes": Array [
              Object {
                "display": true,
                "scaleLabel": Object {
                  "display": true,
                  "labelString": "Values",
                },
                "ticks": Object {
                  "precision": 0,
                  "suggestedMax": 28,
                  "suggestedMin": 17,
                },
              },
            ],
          },
          "tooltips": Object {
            "mode": "nearest",
          },
        }
      }
    />
  </div>
</div>
`;

exports[`<Chart/> Produces a correct chart for category FAVORITE_VIEWS, aggregationLevel AVERAGE, and chartType TOTAL 1`] = `
<div
  className="uua-data-container uua-chart-container"
>
  <h4
    className="uaa-chart-title"
  >
    Favorite views
  </h4>
  <h6
    className="uaa-chart-subtitle"
  >
    Number of times users looked at analysis favorites
  </h6>
  <div
    className="uaa-chart-wrap"
  >
    <Line
      data={
        Object {
          "datasets": Array [
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "borderColor": "#7cb5ec",
              "data": Array [
                104,
                101,
              ],
              "fill": false,
              "label": "Average",
              "lineTension": 0,
            },
          ],
          "labels": Array [
            "Week 22 / 2016",
            "Week 23 / 2016",
          ],
        }
      }
      options={
        Object {
          "animation": Object {
            "duration": 180,
          },
          "legend": Object {
            "position": "bottom",
          },
          "maintainAspectRatio": false,
          "onResize": [Function],
          "responsive": true,
          "scales": Object {
            "xAxes": Array [
              Object {
                "display": true,
                "scaleLabel": Object {
                  "display": true,
                },
                "ticks": Object {
                  "maxRotation": 45,
                  "minRotation": 45,
                },
              },
            ],
            "yAxes": Array [
              Object {
                "display": true,
                "scaleLabel": Object {
                  "display": true,
                  "labelString": "Values",
                },
                "ticks": Object {
                  "precision": 0,
                  "suggestedMax": 104,
                  "suggestedMin": 101,
                },
              },
            ],
          },
          "tooltips": Object {
            "mode": "nearest",
          },
        }
      }
    />
  </div>
</div>
`;

exports[`<Chart/> Produces a correct chart for category FAVORITE_VIEWS, aggregationLevel SUM, and chartType ALL 1`] = `
<div
  className="uua-data-container uua-chart-container"
>
  <h4
    className="uaa-chart-title"
  >
    Favorite views
  </h4>
  <h6
    className="uaa-chart-subtitle"
  >
    Number of times users looked at analysis favorites
  </h6>
  <div
    className="uaa-chart-wrap"
  >
    <Line
      data={
        Object {
          "datasets": Array [
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "borderColor": "#7cb5ec",
              "data": Array [
                4457,
                4093,
              ],
              "fill": false,
              "label": "Map",
              "lineTension": 0,
            },
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "borderColor": "#CC6600",
              "data": Array [
                5047,
                5179,
              ],
              "fill": false,
              "label": "Chart",
              "lineTension": 0,
            },
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "borderColor": "#CCCC00",
              "data": Array [
                4281,
                4275,
              ],
              "fill": false,
              "label": "Pivot Table",
              "lineTension": 0,
            },
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "borderColor": "#66CC00",
              "data": Array [
                4649,
                4856,
              ],
              "fill": false,
              "label": "Event Report",
              "lineTension": 0,
            },
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "borderColor": "#ff0066",
              "data": Array [
                4716,
                4782,
              ],
              "fill": false,
              "label": "Event Chart",
              "lineTension": 0,
            },
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "borderColor": "#000000",
              "data": Array [
                6460,
                6193,
              ],
              "fill": false,
              "label": "Dashboard",
              "lineTension": 0,
            },
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "borderColor": "#00CCCC",
              "data": Array [
                4414,
                4242,
              ],
              "fill": false,
              "label": "Data Set Report",
              "lineTension": 0,
            },
          ],
          "labels": Array [
            "Week 22 / 2016",
            "Week 23 / 2016",
          ],
        }
      }
      options={
        Object {
          "animation": Object {
            "duration": 180,
          },
          "legend": Object {
            "position": "bottom",
          },
          "maintainAspectRatio": false,
          "onResize": [Function],
          "responsive": true,
          "scales": Object {
            "xAxes": Array [
              Object {
                "display": true,
                "scaleLabel": Object {
                  "display": true,
                },
                "ticks": Object {
                  "maxRotation": 45,
                  "minRotation": 45,
                },
              },
            ],
            "yAxes": Array [
              Object {
                "display": true,
                "scaleLabel": Object {
                  "display": true,
                  "labelString": "Values",
                },
                "ticks": Object {
                  "precision": 0,
                  "suggestedMax": 6460,
                  "suggestedMin": 4093,
                },
              },
            ],
          },
          "tooltips": Object {
            "mode": "nearest",
          },
        }
      }
    />
  </div>
</div>
`;

exports[`<Chart/> Produces a correct chart for category FAVORITE_VIEWS, aggregationLevel SUM, and chartType TOTAL 1`] = `
<div
  className="uua-data-container uua-chart-container"
>
  <h4
    className="uaa-chart-title"
  >
    Favorite views
  </h4>
  <h6
    className="uaa-chart-subtitle"
  >
    Number of times users looked at analysis favorites
  </h6>
  <div
    className="uaa-chart-wrap"
  >
    <Line
      data={
        Object {
          "datasets": Array [
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "borderColor": "#7cb5ec",
              "data": Array [
                23669,
                23840,
              ],
              "fill": false,
              "label": "Total",
              "lineTension": 0,
            },
          ],
          "labels": Array [
            "Week 22 / 2016",
            "Week 23 / 2016",
          ],
        }
      }
      options={
        Object {
          "animation": Object {
            "duration": 180,
          },
          "legend": Object {
            "position": "bottom",
          },
          "maintainAspectRatio": false,
          "onResize": [Function],
          "responsive": true,
          "scales": Object {
            "xAxes": Array [
              Object {
                "display": true,
                "scaleLabel": Object {
                  "display": true,
                },
                "ticks": Object {
                  "maxRotation": 45,
                  "minRotation": 45,
                },
              },
            ],
            "yAxes": Array [
              Object {
                "display": true,
                "scaleLabel": Object {
                  "display": true,
                  "labelString": "Values",
                },
                "ticks": Object {
                  "precision": 0,
                  "suggestedMax": 23840,
                  "suggestedMin": 23669,
                },
              },
            ],
          },
          "tooltips": Object {
            "mode": "nearest",
          },
        }
      }
    />
  </div>
</div>
`;

exports[`<Chart/> Produces a correct chart for category FAVORITES_SAVED 1`] = `
<div
  className="uua-data-container uua-chart-container"
>
  <h4
    className="uaa-chart-title"
  >
    Favorites saved
  </h4>
  <h6
    className="uaa-chart-subtitle"
  >
    Number of analysis favorites created by users
  </h6>
  <div
    className="uaa-chart-wrap"
  >
    <Line
      data={
        Object {
          "datasets": Array [
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "borderColor": "#7cb5ec",
              "data": Array [
                1658,
                1593,
              ],
              "fill": false,
              "label": "Maps",
              "lineTension": 0,
            },
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "borderColor": "#CC6600",
              "data": Array [
                1723,
                1668,
              ],
              "fill": false,
              "label": "Charts",
              "lineTension": 0,
            },
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "borderColor": "#CCCC00",
              "data": Array [
                1642,
                1643,
              ],
              "fill": false,
              "label": "Pivot Tables",
              "lineTension": 0,
            },
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "borderColor": "#66CC00",
              "data": Array [
                1748,
                1745,
              ],
              "fill": false,
              "label": "Event Reports",
              "lineTension": 0,
            },
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "borderColor": "#ff0066",
              "data": Array [
                1613,
                1672,
              ],
              "fill": false,
              "label": "Event Charts",
              "lineTension": 0,
            },
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "borderColor": "#000000",
              "data": Array [
                1873,
                1853,
              ],
              "fill": false,
              "label": "Dashboards",
              "lineTension": 0,
            },
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "borderColor": "#00CCCC",
              "data": Array [
                1826,
                1905,
              ],
              "fill": false,
              "label": "Indicators",
              "lineTension": 0,
            },
          ],
          "labels": Array [
            "Week 22 / 2016",
            "Week 23 / 2016",
          ],
        }
      }
      options={
        Object {
          "animation": Object {
            "duration": 180,
          },
          "legend": Object {
            "position": "bottom",
          },
          "maintainAspectRatio": false,
          "onResize": [Function],
          "responsive": true,
          "scales": Object {
            "xAxes": Array [
              Object {
                "display": true,
                "scaleLabel": Object {
                  "display": true,
                },
                "ticks": Object {
                  "maxRotation": 45,
                  "minRotation": 45,
                },
              },
            ],
            "yAxes": Array [
              Object {
                "display": true,
                "scaleLabel": Object {
                  "display": true,
                  "labelString": "Values",
                },
                "ticks": Object {
                  "precision": 0,
                  "suggestedMax": 1905,
                  "suggestedMin": 1593,
                },
              },
            ],
          },
          "tooltips": Object {
            "mode": "nearest",
          },
        }
      }
    />
  </div>
</div>
`;

exports[`<Chart/> Produces a correct chart for category USERS 1`] = `
<div
  className="uua-data-container uua-chart-container"
>
  <h4
    className="uaa-chart-title"
  >
    Users
  </h4>
  <h6
    className="uaa-chart-subtitle"
  >
    Number of users in the system
  </h6>
  <div
    className="uaa-chart-wrap"
  >
    <Line
      data={
        Object {
          "datasets": Array [
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "borderColor": "#7cb5ec",
              "data": Array [
                227,
                234,
              ],
              "fill": false,
              "label": "Active Users",
              "lineTension": 0,
            },
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "borderColor": "#CC6600",
              "data": Array [
                262,
                262,
              ],
              "fill": false,
              "label": "Total Users",
              "lineTension": 0,
            },
          ],
          "labels": Array [
            "Week 22 / 2016",
            "Week 23 / 2016",
          ],
        }
      }
      options={
        Object {
          "animation": Object {
            "duration": 180,
          },
          "legend": Object {
            "position": "bottom",
          },
          "maintainAspectRatio": false,
          "onResize": [Function],
          "responsive": true,
          "scales": Object {
            "xAxes": Array [
              Object {
                "display": true,
                "scaleLabel": Object {
                  "display": true,
                },
                "ticks": Object {
                  "maxRotation": 45,
                  "minRotation": 45,
                },
              },
            ],
            "yAxes": Array [
              Object {
                "display": true,
                "scaleLabel": Object {
                  "display": true,
                  "labelString": "Values",
                },
                "ticks": Object {
                  "precision": 0,
                  "suggestedMax": 262,
                  "suggestedMin": 227,
                },
              },
            ],
          },
          "tooltips": Object {
            "mode": "nearest",
          },
        }
      }
    />
  </div>
</div>
`;
